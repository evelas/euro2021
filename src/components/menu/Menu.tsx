import React from 'react'
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import burger from '../../assets/img/cmsburger.png'
import { authActions } from '../../redux/actions';

const Menu = () => {
  const dispatch = useDispatch();
  const [isOpen, setIsOpen] = React.useState<boolean>(false);
  const handleToggleMenu = () => {
    setIsOpen(!isOpen)
  }
  const setLogout = () => {
    dispatch(authActions.setLogout());
  };
  return (
    <>
      <img id="burger" src={burger} width="22px" height="22px" alt="menu" onClick={handleToggleMenu}/>
      {
        isOpen && (
          <div className="menu">
            <Link to={"/dashboard/home"} className="menu__link">
              <div className="menu__button menu__search">Поиск в системе</div>
            </Link>
            <a href="/cms/excel3" className="menu__link" target="_blank">
              <div className="menu__button menu__cloud">Выгрузка</div>
            </a>
            <Link to={"/dashboard/recruit"} className="menu__link">
              <div className="menu__button menu__recruit">Рекрутинг</div>
            </Link>
            <a className="menu__link" onClick={setLogout}>
              <div className="menu__button">Выйти</div>
            </a>

          </div>
        )
      }

    </>
  )
}

export default Menu
