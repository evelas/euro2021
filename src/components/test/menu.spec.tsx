import { shallow, ShallowWrapper, render } from "enzyme";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import Menu from "../menu/Menu";

const setUp = () => shallow(<Menu setLogout={() => {console.log('test')}} />);

describe("Menu component", () => {
  let component: ShallowWrapper;
  beforeEach(() => {
    component = setUp();
  });
  it("should contain .menu wrapper", () => {
    const wrapper = component.find(".menu");
    expect(wrapper.length).toBe(1);
  });
  it("should render Menu component", () => {
    const component = render(<BrowserRouter><Menu setLogout={() => {console.log('test')}} /></BrowserRouter>);
    expect(component).toMatchSnapshot();
  });

});
