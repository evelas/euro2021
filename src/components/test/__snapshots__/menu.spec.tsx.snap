// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should render Menu component 1`] = `
<div
  class="menu"
>
  <a
    class="menu__link"
    href="/dashboard/home"
  >
    <div
      class="menu__button menu__search"
    >
      Поиск в системе
    </div>
  </a>
  <a
    class="menu__link"
    href="/cms/excel3"
    target="_blank"
  >
    <div
      class="menu__button menu__cloud"
    >
      Выгрузка
    </div>
  </a>
  <a
    class="menu__link"
    href="/dashboard/recruit"
  >
    <div
      class="menu__button menu__recruit"
    >
      Рекрутинг
    </div>
  </a>
  <a
    class="menu__link"
  >
    <div
      class="menu__button"
    >
      Выйти
    </div>
  </a>
</div>
`;
